src/genegraph/sink/stream.clj

```diff
@@ -305,23 +377,41 @@
               (if (< (.position c (first tps)) end)
                 (recur (concat records (poll-once c)))
                 records))
-            (mapv #(consumer-record-to-clj % topic)))))))
+            (mapv #(consumer-record-to-event % topic)))))))
 
-(defn topic-data-to-disk 
-  "Read topic data to disk. Assumes single partition topic."
-  [topic dest]
+(defn- topic-data-to-output
+  "Read topic data to disk. Assumes single partition topic.
+  fn is a single argument function accepting a stream record,
+  and handles the output formatting and file naming of the topic data."
+  [topic fn]
   (with-open [c (consumer-for-topic topic)]
     (let [tps (topic-partitions c topic)
           end (-> (.endOffsets c tps) first val)]
       (.assign c tps)
       (.seekToBeginning c tps)
       (loop [records (poll-once c)]
-        (doseq [r records]
-          (with-open [w (io/output-stream (str dest "/" (name topic) "-" (.offset r)))]
-            (.write w (nippy/freeze (consumer-record-to-clj r topic)))))
+        (doseq [record records]
+          (fn record))
         (when (< (.position c (first tps)) end)
           (recur (poll-once c)))))))
 
+(defn topic-data-to-disk 
+  "Read topic data to disk. Converts the topic record to a genegraph event and writes a
+  nippy compressed version of the entire genegraph event to disk."
+  [topic dest]
+  (topic-data-to-output topic
+                        (fn [record] 
+                          (with-open [w (io/output-stream (str dest "/" (name topic) "-" (.offset record)))]
+                            (.write w (nippy/freeze (consumer-record-to-event record topic)))))))
+
+(defn topic-json-to-disk 
+  "Read topic data, converts to genegraph event, writes the raw value (usually json) to disk."
+  [topic dest]
+  (topic-data-to-output topic
+                        (fn [record]
+                          (->> (consumer-record-to-event record topic)
+                               :genegraph.sink.event/value
+                               (spit (str dest "/" (name topic) "-" (.offset record) ".json"))))))
```

Comment:
The difference between these 2 writers is the file name suffix (`.json`)
and the `::event/value` selection?

```clojure
(defn topic-json-to-disk
  "Read topic data, converts to genegraph event, writes the raw value (usually json) to disk."
  [topic dest]
  (topic-data-to-output
   topic
   (fn [record]
     (let [file (str dest "/" (name topic) "-" (.offset record) ".json")]
       (with-open [w (io/output-stream file)]
         (.write w (::event/value (consumer-record-to-event record topic))))))))
```